src/main/java/ch/njol/skript/aliases/Aliases.java

```diff
@@ -381,17 +356,41 @@ public static void clear() {
 	}
 	
 	/**
-	 * Loads aliases from Skript's standard locations.
+	 * Loads aliases from Skript's standard locations asynchronously.
 	 * Exceptions will be logged, but not thrown.
+	 *
+	 * @return A future that completes when the aliases are loaded.
+	 * The returned value is true if the loading was successful, false otherwise.
 	 */
-	public static void load() {
-		try {
-			long start = System.currentTimeMillis();
-			loadInternal();
-			Skript.info("Loaded " + provider.getAliasCount() + " aliases in " + (System.currentTimeMillis() - start) + "ms");
-		} catch (IOException e) {
-			Skript.exception(e);
-		}
+	public static CompletableFuture<Boolean> load() {
```

Comment:
we could make this a non-breaking change if this signature is left alone and a new loadAsync method is introduced.